### PR TITLE
Add Dockerfile and docker-compose.yaml

### PR DESCRIPTION
**Pull request by julianaklulo**
#### What
Add Dockerfile and docker-compose to build the application and the database.

#### Why
The project didn't have a way to run the application using containers.

### CODE REVIEW

**Review by zsinx6 — APPROVED** [review]

**Review by heitorPB — CHANGES_REQUESTED** [review]
Could you add some notes to the README (or somewhere else) about these?

File: backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+FROM python:3.8 as requirements-stage
+
+RUN apt-get -y update
+RUN apt-get install -y --fix-missing build-essential cmake libpq-dev
```
Comment (heitorPB) [inline backend/Dockerfile]: Why these dependencies are necessary?

Comment (zsinx6) [reply backend/Dockerfile]: postgres

Comment (heitorPB) [reply backend/Dockerfile]: But we are not building the postgres image here.

The lm app starts with a clean `python:3.8` image, right (see line 18 below)? Shouldn't these deps be there instead?

Comment (zsinx6) [reply backend/Dockerfile]: Totally

Comment (julianaklulo) [reply backend/Dockerfile]: I moved the postgres dependencies to the LM image.

**Comment by julianaklulo** [comment]
> Could you add some notes to the README (or somewhere else) about these?

@heitorPB the README seems to be outdated, but I'll include the command to run the backend using `docker-compose`.

**Comment by dusktreader** [comment]
> > Could you add some notes to the README (or somewhere else) about these?
> 
> @heitorPB the README seems to be outdated, but I'll include the command to run the backend using `docker-compose`.

There's a task to update the README as a part of the V3 deliverables: https://app.clickup.com/t/18022949/LM-43.

Adding some info about the docker-compose is fine for now, and we'll circle back to updating the README after V2.1 is finished.

File: backend/Dockerfile (hunk 1 above)
Comment (julianaklulo) [reply backend/Dockerfile]: Actually, we don't need these dependencies, I removed them and it worked.

File: backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+FROM python:3.8
+
+WORKDIR /app
+
+COPY --from=requirements-stage /tmp/requirements.txt /app/requirements.txt
+RUN pip install --no-cache-dir --upgrade -r /app/requirements.txt
+
+
+COPY . /app
+RUN pip install .
```
Comment (heitorPB) [inline backend/Dockerfile:18]: Do we need two `pip install` commands?

Comment (julianaklulo) [reply backend/Dockerfile:18]: The `pip install .` is needed for this: https://github.com/omnivector-solutions/license-manager/blob/9a3ec291d74233550c39a28f41838a74e9c88dc3/backend/lm_backend/main.py#L62

**Pull request merged**